### ISSUE TITLE
Large downloads from file list take long time to start and then time out

### ISSUE THREAD

**Issue by sfisher** [CDL-Dryad/dryad-product-roadmap#444]
I got timeouts for individual file downloads for large files.

This dataset has a lot of large files to download over 8 GB each.  http://dryadx2-prd.cdlib.org/stash/dataset/doi:10.5061/dryad.7pj51v0

I attempted this download http://dryadx2-prd.cdlib.org/stash/downloads/file_stream/84739 and got   "504 Gateway Time-out"

I went into the Merritt collection (info in our dryad tenant config) and tried downloading a large file like one from this page.  https://merritt.cdlib.org/m/ark%253A%252F13030%252Fm5wx2krd/1  

The 8 GB file I tried in Merritt took probably 2 or 3 minutes before it began downloading any bytes.  Then it seemed to work.  I think the only real difference between the Merritt UI and us is that they have a longer UI timeout waiting for the download.  I believe we are essentially using the same rails streaming code as them, also.

It seems like Merritt-Express immediately begins downloading bytes if I try a command like `wget --user=<username> --password=<pwd> http://mrtexpress.cdlib.org/dv/1/ark:/13030/m5wx2krd/14431X10.tar.gz`.

Sadly I don't know if the problem is our code, the apache server in front of passenger or what.

**Comment by sfisher** [comment]
I looked at our code and we are trying to obtain information about the file from Merritt by a head request.  I seem to recall that this takes as long as the full file download from Merritt, so may be the cause of our timeout.

I might be able to populate the download metadata needed for the http request based on what is stored in our database, but it would be better to have a head request that we knows supplies absolutely correct information since we rely heavily on the user and their browser working right to get the mimetype and other things.

I seem to remember this issue coming up in the past and Mark mentioning they could modify the HEAD request stuff in Merritt to be quick and just return info rather than taking as long as a file download to give the information.

I'll try and see if bypassing the HEAD request works and if it starts downloading immediately.

Otherwise, I found the page at https://blog.clearbit.com/rack-csv-streaming/ which might help to see how other streaming methods work, if ours has some flaw.
